templates/template_data.go

```diff
@@ -67,14 +68,25 @@ func removePrivateItems(kv template.KV) template.KV {
 }
 
 func extendAlert(alert template.Alert, externalURL string, logger log.Logger) *ExtendedAlert {
+	//LOGZ.IO GRAFANA CHANGE :: DEV-45466: complete fix switch to account query param functionality
+	accountId := alert.Annotations[models.LogzioAccountIdAnnotation]
+	var generatorUrl string
+	parsedGeneratorUrl, err := receivers.ParseLogzioAppPath(alert.GeneratorURL)
+	if err == nil {
```

Comment:
do you also need to verify accountId is not nil or empty , or is it handled with the `AppendSwitchToAccountQueryParam()` ?

Reply:
i am not sure here is the place to verify it, as it is an async flow and relies simply on previous in memory initialization. (regardless, this verification was not exist in previous grafana. this code was copied.)

Reply:
In general in these kind of validations I don't think the flow or even the logic is something to consider. Because you can always have a bug or some new usage not expected.

Only question is what happens if you don't have the input you expected and how do you want to behave?

If it's OK to throw and fail the entire process for example because of a possible null exception or other illegal operation - it's an option. If it isn't, then I would add some check so we can either handle it or fail more "gracefully" with a log.

Anyway, do whatever you prefer and however you understand the logic here. I just wanted to make sure it is considered